### PR TITLE
Qt: Improve and better integrate updater

### PR DESCRIPTION
**Pull request by spycrab**
*TODO: Screenshots*

### CODE REVIEW

**Comment by spycrab** [comment]
@delroth @JMC47 @MayImilae

File: Source/Core/DolphinQt2/Settings/InterfacePane.cpp [hunk 1]
```diff
@@ -221,5 +247,22 @@ void InterfacePane::OnSaveConfig()
         tr("You must restart Dolphin in order for the change to take effect."));
   }
 
+  QString value;
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Settings/InterfacePane.cpp]: This block can be put into its own function.

File: Source/Core/DolphinQt2/Main.cpp [hunk 2]
```diff
@@ -21,9 +21,9 @@
 #include "DolphinQt2/Resources.h"
 #include "DolphinQt2/Settings.h"
 #include "DolphinQt2/Translation.h"
-#include "UICommon/AutoUpdate.h"
 #include "UICommon/CommandLineParse.h"
 #include "UICommon/UICommon.h"
+#include "Updater.h"
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Main.cpp]: `DolphinQt2/Updater.h`

File: Source/Core/DolphinQt2/Updater.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+#include <QCheckBox>
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QPushButton>
+#include <QTextEdit>
+#include <QVBoxLayout>
+
+#include <iostream>
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Updater.cpp]: `<iostream>` shouldn't be included.

File: Source/Core/DolphinQt2/Updater.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+void Updater::run()
+{
+  if (CheckForUpdate())
+    m_parent->close();
+}
+
+bool Updater::OnUpdateAvailable(const NewVersionInformation& info)
+{
+  std::cout << "Update available!" << std::endl;
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Updater.cpp]: Left in test logging?

File: Source/Core/DolphinQt2/Updater.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
+
+#pragma once
+
+// Copyright 2018 Dolphin Emulator Project
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Updater.h]: Copyright comes before the pragma

**Comment by Helios747** [comment]
Suggestions

put the track selection underneath auto update, and disable the UI element when auto update is set to disabled.

**Comment by delroth** [comment]
Because of the save states issue, I wouldn't actually support "Always update". I would only show a track selector in the config, with options "Don't update", "Stable (once a year)", "Beta (once a month)", "Dev (multiple times a day)".

**Comment by MayImilae** [comment]
Absolutely what delroth said. I can't imagine anyone wanting to update literally every time they run Dolphin, even within the same day! Having the "beta" and "stable" are really important for this feature!

**Comment by Simonx22** [comment]
Updated screenshot: 
image

**Comment by mbc07** [comment]
While this selects the update channel, how the user is supposed to select the behavior, eg. update after closing Dolphin vs update then restart Dolphin?

**Comment by delroth** [comment]
screenshot 2018-03-22 23 51 40

Like that.

**Comment by delroth** [comment]
LGTM. We can iterate on the UI later with more feedback from people.

One small bug: if you click the "never update" button then go immediately to config, the track still shows as "dev".

**Comment by spycrab** [comment]
@delroth Fixed.

**Pull request merged**